src/services/aave.js

```
@@ -0,0 +1,244 @@
+    }
+
+    return this.lendingPoolCoreAddress;
+  }
+
+  async getLendingPoolCoreContract(): Promise<Object> {
+    const lendingPoolCoreAddress = await this.getLendingPoolCoreAddress();
+
+    return Promise.resolve(getContract(
+      lendingPoolCoreAddress,
+      AAVE_LENDING_POOL_CORE_CONTRACT_ABI,
+    ));
+  }
+
+  async getLendingPoolAddress(): Promise<string> {
+    if (!this.lendingPoolAddressesProvider) return '';
+
+    if (!this.lendingPoolAddress) {
+      this.lendingPoolAddress = await this.lendingPoolAddressesProvider.getLendingPool();
+    }
+
+    return this.lendingPoolAddress;
+  }
+
+  async getLendingPoolContract(): Promise<Object> {
+    const lendingPoolAddress = await this.getLendingPoolAddress();
+
+    return Promise.resolve(getContract(
+      lendingPoolAddress,
+      AAVE_LENDING_POOL_CONTRACT_ABI,
+    ));
+  }
+
+  async getAaveTokenAddress(assetAddress: string): Promise<?string> {
+    if (!this.aaveTokenAddresses[assetAddress]) {
+      const lendingPoolCoreContract = await this.getLendingPoolCoreContract();
+      if (!lendingPoolCoreContract) return null;
+      this.aaveTokenAddresses[assetAddress] = await lendingPoolCoreContract.getReserveATokenAddress(assetAddress);
+    }
+    return Promise.resolve(this.aaveTokenAddresses[assetAddress]);
+  }
+
+  async getAaveTokenContractForAsset(assetAddress: string): Promise<?Object> {
+    if (!this.lendingPoolAddressesProvider) return null;
+
+    const aaveTokenAddress = await this.getAaveTokenAddress(assetAddress);
+    if (!aaveTokenAddress) return null;
+
+    return Promise.resolve(getContract(
+      aaveTokenAddress,
+      AAVE_TOKEN_ABI,
+    ));
+  }
+
+  async getAaveTokenAddresses(): Promise<string[]> {
+    const lendingPoolCoreContract = await this.getLendingPoolCoreContract();
+    if (!lendingPoolCoreContract) return [];
+
+    const reserveAddresses = await lendingPoolCoreContract.getReserves().catch(() => []);
```

Just wondering - instead of swallowing errors here (even though the error is handled with `[]`), could we start  capturing the errors in Sentry? We need to increase the visibility of the errors we're encountering to better support the Support team and ourselves:

```suggestion
    const reserveAddresses = await lendingPoolCoreContract.getReserves().catch((e) => {
      Sentry.captureException(e);
      
      return [];
    });
```

FYI: https://docs.sentry.io/platforms/react-native/#capturing-errors